src/main/java/ro/msg/learning/shop/service/locationselection/MostAbundantLocationSelectionAlgorithm.java

```diff
@@ -12,9 +15,11 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+@Component
+@ConditionalOnProperty(name = "ro.msg.learning.shop.locationselect.algorithm", havingValue = "MOST_ABUNDANT")
 public class MostAbundantLocationSelectionAlgorithm implements LocationSelectionAlgorithm {
     @Override
-    public Set<OrderDetail> selectLocationForItems(Collection<OrderDetailWithPotentialLocations> items) {
+    public Set<OrderDetail> selectLocationForItems(Address deliveryAddress, Collection<OrderDetailWithPotentialLocations> items) {
```

Comment:
You should not have an extra parameter for the delivery Address, this way is not extendible at all. If you will add another strategy and need to have another parameter, what would you do? Add another parameter? 

Option1: You can add the deliveryAddress to the OrderDetailWithPotentialLocation or create another object which contains the items and the delivery address. In this way you can add properties to that object in the future without the need to modify all the methods for the other strategies. 

Option2: You can pass as a parameter to this method an Order object which contains the address and the orderDetails (product id and quantity), exactly what you need for the strategies.